### PR TITLE
Resolve flake8 violations in tests/

### PR DESCRIPTION
**Pull request by cjmayo**
Run black and fix the remaining violations.

### CODE REVIEW

**Comment by anarcat** [comment]
maybe this should also enforce black or at least pep8 in CI since we seem to have completely converted the codebase...

File: tests/checker/__init__.py [hunk 1]
```diff
@@ -90,42 +99,44 @@ def log_url(self, url_data):
...
+        if self.has_part("realurl"):
             self.result.append("real url %s" % url_data.url)
-        if self.has_part('name') and url_data.name:
+        if self.has_part("name") and url_data.name:
             self.result.append("name %s" % url_data.name)
-        if self.has_part('base') and url_data.base_ref:
+        if self.has_part("base") and url_data.base_ref:
             self.result.append("baseurl %s" % url_data.base_ref)
-        if self.has_part('info'):
+        if self.has_part("info"):
```
Comment (mgedmin) [inline tests/checker/__init__.py:114]: I hate the noise Black creates by changing the quote characters on all the strings.  And they make the diff so huge.  I'm sure I'll give up somewhere in the middle.

File: tests/checker/__init__.py [hunk 2]
```diff
@@ -210,16 +227,20 @@ def get_resultlines(self, filename):
...
+            "curdir": get_file_url(os.getcwd()),
+            "datadir": get_file_url(get_file()),
+        }
         # the webserver uses the first free port number
-        if hasattr(self, 'port'):
-            d['port'] = self.port
+        if hasattr(self, "port"):
+            d["port"] = self.port
         # all result files are encoded in utf-8
         with codecs.open(resultfile, "r", "utf-8") as f:
```
Comment (mgedmin) [inline tests/checker/__init__.py:238]: Noticed in passing, not part of this PR, but: with the Python 2 drop we can replace `codecs.open(..., "UTF-8")` with just `open(..., encoding="UTF-8")`.

File: tests/checker/httpserver.py [hunk 3]
```diff
@@ -49,10 +49,11 @@ def log_message(self, format, *args):
...
         pass
 
+
 # serve .xhtml files as application/xhtml+xml
-StoppableHttpRequestHandler.extensions_map.update({
-        '.xhtml': 'application/xhtml+xml',
-})
+StoppableHttpRequestHandler.extensions_map.update(
+    {".xhtml": "application/xhtml+xml"}
+)
```
Comment (mgedmin) [inline tests/checker/httpserver.py:56]: Did Black do this?  Weird.  I'd've gone with

```suggestion
StoppableHttpRequestHandler.extensions_map.update({
    ".xhtml": "application/xhtml+xml",
})
```

File: tests/checker/test_httpbin.py [hunk 4]
```diff
@@ -55,11 +55,7 @@ def test_basic_auth(self):
         password = "testpassword"
         url = get_httpbin_url("/basic-auth/%s/%s" % (user, password))
         nurl = self.norm(url)
-        entry = dict(
-            user=user,
-            password=password,
-            pattern=re.compile(r'.*'),
-        )
+        entry = dict(user=user, password=password, pattern=re.compile(r".*"),)
```
Comment (mgedmin) [inline tests/checker/test_httpbin.py:58]:
```suggestion
        entry = dict(user=user, password=password, pattern=re.compile(r".*"))
```

Comment (mgedmin) [reply tests/checker/test_httpbin.py:58]: This is a Black bug BTW: https://github.com/psf/black/issues/1202

(it's closed as duplicate, or, rather, one small instance of the larger bug https://github.com/psf/black/issues/1288)

File: tests/checker/test_loginurl.py [hunk 5]
```diff
@@ -34,19 +35,23 @@ def visit_loginurl(self, page, user=None, password=None, extrafields=False):
...
-        confargs["authentication"] = [{
-            "user": user, "password": password,
-            "pattern": re.compile("^http://localhost.*")
-        }]
+        confargs["authentication"] = [
+            {
+                "user": user,
+                "password": password,
+                "pattern": re.compile("^http://localhost.*"),
+            }
```
Comment (mgedmin) [inline tests/checker/test_loginurl.py:43]:
```suggestion
            },
```

File: tests/checker/test_news.py [hunk 6]
```diff
@@ -25,11 +25,13 @@
 # to enable the has_newsserver() resource manually.
 NNTP_SERVER = "news.uni-stuttgart.de"
 # info string returned by news server
-NNTP_INFO = "200 news.uni-stuttgart.de InterNetNews NNRP server " \
-            "INN 2.5.2 ready (no posting)"
+NNTP_INFO = (
+    "200 news.uni-stuttgart.de InterNetNews NNRP server " "INN 2.5.2 ready (no posting)"
```
Comment (mgedmin) [inline tests/checker/test_news.py:29]:
```suggestion
    "200 news.uni-stuttgart.de InterNetNews NNRP server INN 2.5.2 ready (no posting)"
```

If this is not a Black bug, it should be!

Comment (mgedmin) [reply tests/checker/test_news.py:29]: https://github.com/psf/black/issues/26

Fixed 21 days ago.

Last release in 2019.

*sigh*

File: tests/test_clamav.py [hunk 7]
```diff
@@ -37,25 +36,27 @@ def testClean(self):
...
         )
         infected, errors = clamav.scan(data, self.clamav_conf)
         # different versions of clamav report different responses, apparently
         acceptable_responses = (
-            ['stream: ClamAV-Test-File(2d1206194bd704385e37000be6113f73:781) FOUND\n'],
-            ['stream: Clamav.Test.File-6(aa15bcf478d165efd2065190eb473bcb:544) FOUND\n'],
+            ["stream: ClamAV-Test-File(2d1206194bd704385e37000be6113f73:781) FOUND\n"],
+            [
+              "stream: Clamav.Test.File-6(aa15bcf478d165efd2065190eb473bcb:544) FOUND\n"
+            ],
```
Comment (mgedmin) [inline tests/test_clamav.py:59]: Ugh, this is why I don't like automatic formatters.  They get things right in 99% of the cases, and then throw out something like this.
```suggestion
            [
              "stream: ClamAV-Test-File(2d1206194bd704385e37000be6113f73:781) FOUND\n"
            ],
            [
              "stream: Clamav.Test.File-6(aa15bcf478d165efd2065190eb473bcb:544) FOUND\n"
            ],
```

File: tests/test_po.py [hunk 8]
```diff
@@ -55,13 +56,15 @@ class TestGTranslator(unittest.TestCase):
...
 
     def check_file(self, fd, f):
         """Test for GTranslator broken syntax."""
         for line in fd:
             if line.strip().startswith(b"#"):
                 continue
-            self.assertFalse(b"\xc2\xb7" in line,
-                 "Broken GTranslator copy/paste in %r:\n%r" % (f, line))
+            self.assertFalse(
+                b"\xc2\xb7" in line,
```
Comment (mgedmin) [inline tests/test_po.py:68]: Unrelated to this PR, but perhaps we should be checking for Unicode `\N{MIDDLE DOT}` in decoded text, not the UTF-8 encoding of it in byte strings.

File: tests/test_url.py [hunk 9]
```diff
@@ -95,14 +100,15 @@ def url_quote(url):
...
+            "http://groups.google.com/groups?hl=en&lr&ie=UTF-8&"
+            "threadm=3845B54D.E546F9BD%40monmouth.com&rnum=2&"
+            "prev=/groups%3Fq%3Dlogitech%2Bwingman%2Bextreme%2Bdigital"
+            "%2B3d%26hl%3Den%26lr%3D%26ie%3DUTF-8%26selm%3D3845B54D.E5"
+            "46F9BD%2540monmouth.com%26rnum%3D2"
+        )
         self.urlnormtest(url, url)
-        url = "http://redirect.alexa.com/redirect?"\
-              "http://www.offeroptimizer.com"
+        url = "http://redirect.alexa.com/redirect?" "http://www.offeroptimizer.com"
```
Comment (mgedmin) [inline tests/test_url.py:111]:
```suggestion
        url = "http://redirect.alexa.com/redirect?http://www.offeroptimizer.com"
```

File: tests/cache/test_urlqueue.py [hunk 10]
```diff
@@ -109,9 +102,7 @@ def test_put_has_result_true(self):
         urldata = UrlData(
             url="Bar",
             cache_url="Bar",
-            aggregate=Aggregate(
-                result_cache=self.result_cache,
-            ),
+            aggregate=Aggregate(result_cache=self.result_cache,),
```
[the remark this answers is not in the document]
Comment (mgedmin) [reply tests/cache/test_urlqueue.py:105]: Well, "many" ~= 6 or 7, in this file, all about `Aggregate(...,)`, so should be easy to search-and-replace fix.

[the remark this answers is not in the document]
Comment (mgedmin) [reply tests/cache/test_urlqueue.py:105]: `git grep ',)'` confirms that there are Aggregates here and one dict literal in

```
tests/checker/test_httpbin.py:        entry = dict(user=user, password=password, pattern=re.compile(r".*"),)
```

that have the bad trailing comma right in front of the closing parenthesis.  All other trailing commas like this are in 1-tuples, where they belong.

**Comment by mgedmin** [comment]
> maybe this should also enforce black

Let's not, at least not until Black gets the known bugs fixed and released.

> or at least pep8 in CI since we seem to have completely converted the codebase...

tox -e flake8 complains a lot about code in linkcheck/.

I've also noticed that tox -e flake8 complains about code in `build/`, which it should not do.

File: tests/test_url.py (hunk 9 above)
Comment (mgedmin) [reply tests/test_url.py:111]: `git grep '" "'` tells me I missed another one like this, in this file:
```python
url = "http://server/..%5c..%5c..%5c..%5c..%5c..%5c..%5c.." "%5ccskin.zip"
```

File: tests/checker/__init__.py (hunk 1 above)
Comment (anarcat) [reply tests/checker/__init__.py:114]: i also find this change annoying, because i usually type single quotes because they are faster (one keystroke instead of shift+keystroke) on my keyboard.

the black docs have a section on this:

https://black.readthedocs.io/en/stable/the_black_code_style.html#strings

their argument ias that double-quotes are less ambiguous and that you can always type single quotes and let black fix the formatting if you don't like to type double-quotes, which is reasonable.

That said, there's an escape hatch in black for this:

> If you are adopting Black in a large project with pre-existing string conventions (like the popular “[single quotes for data, double quotes for human-readable strings](https://stackoverflow.com/a/56190)”), you can pass `--skip-string-normalization` on the command line. This is meant as an adoption helper, avoid using this for new projects.

Do we want to do that?

File: tests/checker/httpserver.py (hunk 3 above)
Comment (ichard26) [reply tests/checker/httpserver.py:56]: Hi, _Black_ contributor here. Yeah, _Black_ did that, _Black_ will unpack multiple brackets. 

Pulled from https://github.com/psf/black/issues/177#issuecomment-385201073
> As for unpacking multiple brackets to separate indentation level, Black will continue doing this. I don't like multiple unrelated brackets shoved together to save a line.

File: tests/checker/__init__.py (hunk 1 above)
Comment (mgedmin) [reply tests/checker/__init__.py:114]: Yes please, anything for a smaller diff.

(I won't mind if there's a subsequent PR thst does only quote changed, separately, if anyone wants to do that.  Personally, I'm not yet convinced by Black's arguments about quotes.)

File: tests/checker/httpserver.py (hunk 3 above)
Comment (mgedmin) [reply tests/checker/httpserver.py:56]: Ah, thank you for clarifying!

I like multiple unrelated brackets shoved together, not to save a line, but to save an indentation level.

File: tests/cache/test_urlqueue.py [hunk 11]
```diff
@@ -89,9 +84,7 @@ def test_put_has_result_false(self):
...
```
[the remark this answers is not in the document]
Comment (cjmayo) [reply tests/cache/test_urlqueue.py:87]: Done.

File: tests/cache/test_urlqueue.py (hunk 10 above)
[the remark this answers is not in the document]
Comment (cjmayo) [reply tests/cache/test_urlqueue.py:105]: Another manual check to add to the process.

File: tests/checker/__init__.py [hunk 12]
```diff
@@ -58,27 +58,36 @@ def __init__(self, **kwargs):
...
-            sorted(['url %s' % x.strip() for x in
-                    re.split(r'^url .*?', '\n'.join(result_log),
-                             flags=re.DOTALL | re.MULTILINE)
-                    if x])).splitlines()
+        return "\n".join(
+            sorted(
+                [
+                    "url %s" % x.strip()
+                    for x in re.split(
+                        r"^url .*?",
```
[the remark this answers is not in the document]
Comment (cjmayo) [reply tests/checker/__init__.py:83]: Done.

File: tests/checker/test_loginurl.py (hunk 5 above)
Comment (cjmayo) [reply tests/checker/test_loginurl.py:43]: Done.

File: tests/checker/test_news.py (hunk 6 above)
Comment (cjmayo) [reply tests/checker/test_news.py:29]: Done.

File: tests/test_clamav.py (hunk 7 above)
Comment (cjmayo) [reply tests/test_clamav.py:59]: Done

File: tests/test_url.py (hunk 9 above)
Comment (cjmayo) [reply tests/test_url.py:111]: Done.

**Comment by cjmayo** [comment]
Added commits in response to the review comments.
